src/addrman.cpp

```diff
@@ -366,8 +370,11 @@ CAddrInfo CAddrMan::Select_()
             if (GetRandInt(1 << 30) < fChanceFactor * info.GetChance() * (1 << 30))
                 return info;
             fChanceFactor *= 1.2;
+            MilliSleep(100);
         }
     }
+    
```

Comment:
Sorry, my comment was criticising the original code, but I didn't make that particularly clear. Thanks for your contribution!